### PR TITLE
Remove key parser and take I/O as string

### PR DESCRIPTION
**Pull request by divya2661**

### CODE REVIEW

File: example/main.go [hunk 1]
```diff
@@ -25,9 +24,10 @@ type keyValue struct {
 
 func main() {
 	flag.Parse()
-
-	gossipDb, _ := gossipdb.NewGossipDb(*members, *rpc_port, &keyParser{})
-
+	gossipDb, err := gossipdb.NewGossipDb(*members, *rpc_port)
+	if err != nil {
+		fmt.Println(err)
```
Comment (SinisterLight) [inline example/main.go]: panic on this error ?

**Review by SinisterLight — CHANGES_REQUESTED** [review]

File: gossipdb.go [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+	"encoding/json"
+	"fmt"
 	"github.com/hashicorp/memberlist"
 )
 
-type KeyParser interface {
-	ToKey(b []byte) string
+type Pair struct {
+	Key   string
+	Value string
```
Comment (SinisterLight) [inline gossipdb.go]: How about allowing `value` to be `interface{}`, honestly i see no constraint yet on this ?

File: example/main.go [hunk 3]
```diff
@@ -45,7 +45,10 @@ func addKeyHandler(g *gossipdb.GossipDb) http.HandlerFunc {
 		if err != nil {
 			fmt.Println(err)
 		}
-		g.Set(body)
+		var data keyValue
```
Comment (SinisterLight) [inline example/main.go:48]: How about killing `keyValue` struct and using `Pair`struct declared below directly ?

File: gossipdb.go [hunk 4]
```diff
@@ -45,19 +48,27 @@ func NewGossipDb(members string, port int, k KeyParser) (*GossipDb, error) {
...
+	if found {
+		return value, found
+	}
+	return "nil", found
 }
 
-func (gdb *GossipDb) Set(m []byte) {
-	k := gdb.keyParser.ToKey(m)
-	gdb.database.Save(k, m)
+func (gdb *GossipDb) Set(key string, value string) {
```
Comment (SinisterLight) [inline gossipdb.go]: This interface can also take in `Pair` ?

**Comment by SinisterLight** [comment]
Overall its great, wanted to change that `keyParser` monstrosity for a long time, thanks!

File: example/main.go (hunk 1 above)
Comment (divya2661) [reply example/main.go]: Yes, It should. Will do

File: example/main.go (hunk 3 above)
Comment (divya2661) [reply example/main.go:48]: I think keeping different struct will reduce confusions. So we can keep it for the time being.

File: gossipdb.go (hunk 2 above)
Comment (divya2661) [reply gossipdb.go]: Yup, I was Also thinking the same. Will change it to interface

**Review by SinisterLight — APPROVED** [review]

**Pull request merged**